node/node_test.go

```diff
@@ -18,67 +18,89 @@ import (
 
 var MockServerAddr = ":7980"
 
+// NodeType defines the type of a node using iota for an enumerated type.
+type NodeType int
+
+const (
+	// FullNode represents a full node type.
+	FullNodeType NodeType = iota
+	// LightNode represents a light node type.
+	LightNodeType
+)
```

Comment:
looks like these changes fell in to this PR. let's remove.

The impact of this change is lost in the diff of the `startNodeWithCleanup` changes. 
I'd rather hold of on this to be in a separate PR to better understand it's impact on these changes too
```
ln := initAndStartNodeWithCleanup(ctx, t, LightNodeType).(*LightNode)
```